src/task/ReceiveVideoStreamIndexTask.ts

```diff
@@ -74,13 +75,38 @@ export default class ReceiveVideoStreamIndexTask extends BaseTask
       videoUplinkBandwidthPolicy,
     } = this.context;
 
+    const oldAttendees = videoStreamIndex.allVideoSendingAttendeesExcludingSelf(selfAttendeeId);
     videoStreamIndex.integrateIndexFrame(indexFrame);
     videoDownlinkBandwidthPolicy.updateIndex(videoStreamIndex);
     videoUplinkBandwidthPolicy.updateIndex(videoStreamIndex);
 
     this.resubscribe(videoDownlinkBandwidthPolicy, videoUplinkBandwidthPolicy);
     this.updateVideoAvailability(indexFrame);
     this.handleIndexVideosPausedAtSource();
+    const newAttendees = videoStreamIndex.allVideoSendingAttendeesExcludingSelf(selfAttendeeId);
+    if (!this.areVideoSendingAttendeesEqual(oldAttendees, newAttendees)) {
+      this.context.audioVideoController.forEachObserver((observer: AudioVideoObserver) => {
+        Maybe.of(observer.remoteVideosAvailableDidChange).map(f => f.bind(observer)(newAttendees));
+      });
+    }
+  }
+
+  private areVideoSendingAttendeesEqual(
```

Comment:
Nice! You can optimize this method by moving the length check to top.

```
): boolean {
  if (oldAttendees.length !== newAttendees.length) {
     return false;
  }

  // Now, you can avoid two sort() calls!
  ...
```

Reply:
Right, right updated length check before sort. Thanks!